src/libsystemd/sd-network/sd-network.c

```diff
@@ -171,6 +171,17 @@ _public_ int sd_network_link_get_operational_state(int ifindex, char **state) {
         return network_link_get_string(ifindex, "OPER_STATE", state);
 }
 
+_public_ int sd_network_link_get_required_for_online(int ifindex) {
+        _cleanup_free_ char *s = NULL;
+        int r;
+
+        r = network_link_get_string(ifindex, "REQUIRED_FOR_ONLINE", &s);
+        if (r < 0)
+                return true;
```

Comment:
please don't eat up errors. Please propagate r properly. Except if it is -ENODATA (i.e. when the field is missing), return true then.